Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -0,0 +1,687 @@
+        self.ipk = ipk
+        self.ispec = ispec  # spectrum at center of kernel that corresponds to max I/sigma from convolution
+        self.tof = tof  # TOF at center of kernel that corresponds to max I/sigma from convolution
+        self.tof_fwhm = tof_fwhm
+        self.ipks_near = ipks_near
+
+
+class ShoeboxResult:
+    """
+    This class is used to hold data and integration parameters for single-crystal Bragg peaks
+    """
+
+    def __init__(self, ipk, pk, x, y, peak_shape, ipos, ipk_pos, intens_over_sig, status, strong_peak=None, ipk_strong=None):
+        self.peak_shape = list(peak_shape)
+        self.kernel_shape = list(get_kernel_shape(*self.peak_shape)[0])
+        self.ipos = list(ipos) if ipos is not None else ipos
+        self.ipos_pk = list(ipk_pos)
+        self.labels = ["Row", "Col", "TOF"]
+        self.ysum = []
+        self.xmin, self.xmax = np.round(x[0], 1), np.round(x[-1], 1)
+        self.status = status
+        # integrate y over each dim
+        for idim in range(len(peak_shape)):
+            self.ysum.append(y.sum(axis=idim))
+        # extract peak properties
+        intens_over_sig = np.round(intens_over_sig, 1)
+        hkl = np.round(pk.getHKL(), 2)
+        wl = np.round(pk.getWavelength(), 2)
+        tth = np.round(np.degrees(pk.getScattering()), 1)
+        d = np.round(pk.getDSpacing(), 2)
+        self.title = (
+            f"{ipk} ({','.join(str(hkl)[1:-1].split())})"
+            f"\n$I/\\sigma$={intens_over_sig}\n"
+            rf"$\lambda$={wl} $\AA$; "
+            rf"$2\theta={tth}^\circ$; "
+            rf"d={d} $\AA$"
+            f"\n{status.value}"
+        )
+        if strong_peak is not None and ipk_strong is not None:
+            self.title = f"{self.title} (shoebox taken from ipk={ipk_strong} {np.round(strong_peak.getHKL(), 2)})"
+
+    def plot_integrated_peak(self, fig, axes, norm_func, rect_func):
+        for iax, ax in enumerate(axes):
+            im = ax.imshow(self.ysum[iax], aspect=self.ysum[iax].shape[1] / self.ysum[iax].shape[0])
+            im.set_norm(norm_func())
+            # plot shoebox
+            if self.ipos is not None:
+                self.plot_shoebox(iax, ax, rect_func)
+            # plot peak position
+            cen = self.ipos_pk.copy()
+            cen.pop(iax)
+            ax.plot(cen[1], cen[0], "xr")
+            # set labels
+            labels = self.labels.copy()
+            labels.pop(iax)
+            ax.set_xlabel(labels[1])
+            ax.set_ylabel(labels[0])
+        # overwrite axes tick labels for TOF axes
+        nticks = 4
+        xticklabels = nticks * [""]
+        xticklabels[0] = str(self.xmin)
+        xticklabels[-1] = str(self.xmax)
+        for ax in axes[:-1]:
+            ax.set_xticks(np.linspace(*ax.get_xlim(), nticks))
+            ax.set_xticklabels(xticklabels)
+        # set title
+        fig.suptitle(self.title)
+
+    def plot_shoebox(self, iax, ax, rect_func):
+        cen = self.ipos.copy()
```

Comment:
In this class, can `cen` be renamed to `center_peak_pos` or something like that?